src/Analyser/ConstantResolver.php

```diff
@@ -35,12 +39,13 @@ final class ConstantResolver
 
 	/**
 	 * @param string[] $dynamicConstantNames
+	 * @param int|array{min: int, max: int}|null $phpVersion
 	 */
 	public function __construct(
 		private ReflectionProviderProvider $reflectionProviderProvider,
 		private array $dynamicConstantNames,
-		private ?PhpVersion $composerMinPhpVersion,
-		private ?PhpVersion $composerMaxPhpVersion,
+		private int|array|null $phpVersion,
+		private ?ComposerPhpVersionFactory $composerPhpVersionFactory,
```

Comment:
Why is this nullable?

Reply:
to allow [creation of the ConstantResolver](https://github.com/phpstan/phpstan-src/blob/38630336ab201cb9ecce6d4104ec1ea5b08e8b7e/src/DependencyInjection/ValidateIgnoredErrorsExtension.php#L68) when the DI is not yet constructed

Reply:
Just pass `new ComposerPhpVersionFactory` with empty `$composerAutoloaderProjectPaths` there.